cookbook/entities/datagrid.rst

```diff
@@ -0,0 +1,201 @@
+                        priority.label AS taskPriority
+                    from:
+                        - { table: AppBundle:Task, alias: task }
+                    join:
+                        left:
+                            - { join: task.priority, alias: priority }
+
+Displayed Columns
+~~~~~~~~~~~~~~~~~
+
+Then, the ``columns`` option needs to be used to configure how which data will be displayed:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            # ...
+            columns:
+                id:
+                    label: ID
+                    frontend_type: integer
+                subject:
+                    label: Subject
+                description:
+                    label: Description
+                dueDate:
+                    label: Due Date
+                    frontend_type: datetime
+                taskPriority:
+                    label: Priority
+
+For each column, the ``frontend_type`` option can be used to customize how data will be displayed
+(by default, the data will be shown as is).
+
+Column Sorters
+~~~~~~~~~~~~~~
+
+Use the ``sorters`` option to define on which columns' header the user can click to order by the
+data:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            # ...
+            sorters:
+                columns:
+                    id:
+                        data_name: task.id
+                    subject:
+                        data_name: task.subject
+                    description:
+                        data_name: task.description
+                    dueDate:
+                        data_name: task.dueDate
+                    taskPriority:
+                        data_name: priority.label
+                default:
+                    dueDate: DESC
+
+Each key under ``sorters.columns`` refers to one of the displayed columns. The ``data_name`` option
+is the term that will be used as the ``order by`` term in the Doctrine query.
+
+Data Filters
+~~~~~~~~~~~~
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            # ...
+            filters:
+                columns:
+                    id:
+                        type: number
+                        data_name: task.id
+                    subject:
+                        type: string
+                        data_name: task.subject
+                    description:
+                        type: string
+                        data_name: task.description
+                    dueDate:
+                        type: datetime
+                        data_name: task.dueDate
+                    taskPriority:
+                        type: string
+                        data_name: priority.label
+
+.. _cookbook-entities-grid-controller:
+
```

Comment:
Please add here the final `datagrid.yml` file structure. It can be the link to yml file.